### ISSUE TITLE
EM fields object inconsistancy

### ISSUE THREAD

**Issue by micmitch** [simpeg/simpeg#366]
Using the EM.FDEM.Problem3D_h problem class to test analytic expressions for the harmonic electric dipole against numeric expressions. In doing this I realized that there is a significant difference between some of the fields if I calculate them "manually" or fetch them from the fields object. 

`# Define the source
# Search over z-faces to find face nearest src_loc then add nFx to get to global face index [nFx][nFy = 0][nFz]

s_ind = Utils.closestPoints(mesh, src_loc, 'Fz') + mesh.nFx
de = np.zeros(mesh.nF, dtype=complex)
de[s_ind] = 1./csz
de_p = [EM.FDEM.Src.RawVec_e([], freq, de/mesh.area)]
# Pair the problem and survey

survey = EM.FDEM.Survey(de_p)

mapping = [('sigma', Maps.IdentityMap(mesh)), ('mu', Maps.IdentityMap(mesh))]

problem = EM.FDEM.Problem3D_h(mesh, mapping=mapping)
# pair problem and survey

problem.pair(survey)

try:
    from pymatsolver import MumpsSolver
    problem.Solver = MumpsSolver
except ImportError, e:
    problem.Solver = SolverLU
# solve

numFields_ElecDipole = problem.fields(np.r_[SigmaBack, MuBack])

rlim = [20., 500.]
r = mesh.vectorCCx[np.argmin(np.abs(mesh.vectorCCx-rlim[0])):np.argmin(np.abs(mesh.vectorCCx-rlim[1]))]
z = 100.
# where we choose to measure

XYZ = Utils.ndgrid(r, np.r_[0.], np.r_[z])

Pf = mesh.getInterpolationMat(XYZ, 'CC')
Zero = sp.csr_matrix(Pf.shape)
Pfx, Pfz = sp.hstack([Pf, Zero]), sp.hstack([Zero, Pf])
# J lives on faces

j_num = numFields_ElecDipole[de_p, 'j']
Rho = Utils.sdiag(1./SigmaBack)
Rho = sp.block_diag([Rho, Rho])
e_numTest = Rho_mesh.aveF2CCV_j_num
j_num = mesh.aveF2CCV*j_num
# E lives on cell centres

e_num = numFields_ElecDipole[de_p, 'e']
# H lives on edges

h_num = numFields_ElecDipole[de_p, 'h']
h_num = mesh.aveE2CCV*h_num
# B lives on cell centers

b_num = numFields_ElecDipole[de_p, 'b']
Mu = Utils.sdiag(MuBack)
b_numTest = Mu*h_num
# Interpolate numeric fields and fluxes to cell cetres for easy comparison with analytics

ex_num, ez_num = Pfx_e_num, Pfz_e_num
ey_num = np.zeros_like(ex_num)
ex_numTest, ez_numTest = Pfx_e_numTest, Pfz_e_numTest
ey_numTest = np.zeros_like(ex_numTest)
# Check E values computed from fields object

print ' comp,       fields obj,       manual,       fields - manual,       (fields - manual)/manual'
print '  E_x:', np.linalg.norm(ex_num), np.linalg.norm(ex_numTest), np.linalg.norm(ex_num-ex_numTest), np.linalg.norm(ex_num-ex_numTest)/np.linalg.norm(ex_numTest) 
print ''

jx_num, jz_num = Pfx_j_num, Pfz_j_num
jy_num = np.zeros_like(jx_num)
# Since we are evaluating along the plane y=0 the b_theta == b_y in cartesian coordiantes

btheta_num = Pf*b_num
bx_num = np.zeros_like(btheta_num)
bz_num = np.zeros_like(btheta_num)

btheta_numTest = Pf*b_numTest
bx_numTest = np.zeros_like(btheta_numTest)
bz_numTest = np.zeros_like(btheta_numTest)
# Check E values computed from fields object

print ' comp,       fields obj,       manual,       fields - manual,       (fields - manual)/manual'
print '  B_theta:', np.linalg.norm(btheta_num), np.linalg.norm(btheta_numTest), np.linalg.norm(btheta_num - btheta_numTest), np.linalg.norm(btheta_num-btheta_numTest)/np.linalg.norm(btheta_numTest) 
print ''

htheta_num = Pf*h_num
hx_num = np.zeros_like(htheta_num)
hz_num = np.zeros_like(htheta_num)
# get analytic solutions

exa, eya, eza = EM.Analytics.FDEMDipolarfields.E_from_ElectricDipoleWholeSpace(XYZ, src_loc, sigmaback, Utils.mkvc(np.array(freq)),orientation='Z',kappa= kappa)
exa, eya, eza = Utils.mkvc(exa,2), Utils.mkvc(eya,2), Utils.mkvc(eza,2)

jxa, jya, jza = EM.Analytics.FDEMDipolarfields.J_from_ElectricDipoleWholeSpace(XYZ, src_loc, sigmaback, Utils.mkvc(np.array(freq)),orientation='Z',kappa= kappa)
jxa, jya, jza = Utils.mkvc(jxa,2), Utils.mkvc(jya,2), Utils.mkvc(jza,2)

bxa, bya, bza = EM.Analytics.FDEMDipolarfields.B_from_ElectricDipoleWholeSpace(XYZ, src_loc, sigmaback, Utils.mkvc(np.array(freq)),orientation='Z',kappa= kappa)
bxa, bya, bza = Utils.mkvc(bxa,2), Utils.mkvc(bya,2), Utils.mkvc(bza,2)

hxa, hya, hza = EM.Analytics.FDEMDipolarfields.H_from_ElectricDipoleWholeSpace(XYZ, src_loc, sigmaback, Utils.mkvc(np.array(freq)),orientation='Z',kappa= kappa)
hxa, hya, hza = Utils.mkvc(hxa,2), Utils.mkvc(hya,2), Utils.mkvc(hza,2)

print ' comp,       anayltic,       numeric,       num - ana,       (num - ana)/ana'
print '  E_x:', np.linalg.norm(exa), np.linalg.norm(ex_numTest), np.linalg.norm(exa-ex_numTest), np.linalg.norm(exa-ex_numTest)/np.linalg.norm(exa)
print '  E_y:', np.linalg.norm(eya), np.linalg.norm(ey_numTest), np.linalg.norm(eya-ey_numTest)
print '  E_z:', np.linalg.norm(eza), np.linalg.norm(ez_numTest), np.linalg.norm(eza-ez_numTest), np.linalg.norm(eza-ez_numTest)/np.linalg.norm(eza)
print ''
print '  J_x:', np.linalg.norm(jxa), np.linalg.norm(jx_num), np.linalg.norm(jxa-jx_num), np.linalg.norm(jxa-jx_num)/np.linalg.norm(jxa)
print '  J_y:', np.linalg.norm(jya), np.linalg.norm(jy_num), np.linalg.norm(jya-jy_num)
print '  J_z:', np.linalg.norm(jza), np.linalg.norm(jz_num), np.linalg.norm(jza-jz_num), np.linalg.norm(jza-jz_num)/np.linalg.norm(jza)
print ''
print '  H_x:', np.linalg.norm(hxa), np.linalg.norm(hx_num), np.linalg.norm(hxa-hx_num)
print '  H_y:', np.linalg.norm(hya), np.linalg.norm(htheta_num), np.linalg.norm(hya-htheta_num), np.linalg.norm(hya-htheta_num)/np.linalg.norm(hya)
print '  H_z:', np.linalg.norm(hza), np.linalg.norm(hz_num), np.linalg.norm(hza-hz_num)
print ''
print '  B_x:', np.linalg.norm(bxa), np.linalg.norm(bx_num), np.linalg.norm(bxa-bx_num)
print '  B_y:', np.linalg.norm(bya), np.linalg.norm(btheta_numTest), np.linalg.norm(bya-btheta_numTest), np.linalg.norm(bya-btheta_numTest)/np.linalg.norm(bya)
print '  B_z:', np.linalg.norm(bza), np.linalg.norm(bz_num), np.linalg.norm(bza-bz_num)
`

Comparing the E and B fields from field object vs. manual calculation from J and H show that the values differ by about 50%

 comp,       fields obj,       manual,       fields - manual,       (fields - manual)/manual
  E_x: 3.95648139543e-07 2.6355108803e-07 1.32102146212e-07 0.501239236762

 comp,       fields obj,       manual,       fields - manual,       (fields - manual)/manual
  B_theta: 1.41758683016e-11 9.43866212262e-12 4.73772034803e-12 0.501948293782

The manual calculations agree nicely with the analytics.

 comp,       anayltic,       numeric,       num - ana,       (num - ana)/ana
  E_x: 2.61961397615e-07 2.6355108803e-07 2.13770593666e-09 0.00816038529387
  E_y: 0.0 0.0 0.0
  E_z: 3.42014156382e-07 3.43346089763e-07 2.60249101794e-09 0.00760930788793

  J_x: 2.61961397615e-07 2.6355108803e-07 2.13770593666e-09 0.00816038529387
  J_y: 0.0 0.0 0.0
  J_z: 3.42014156382e-07 3.43346089763e-07 2.60249101794e-09 0.00760930788793

  H_x: 0.0 0.0 0.0
  H_y: 7.50861759251e-06 7.51104866495e-06 3.15377969514e-08 0.0042002134964
  H_z: 0.0 0.0 0.0

  B_x: 0.0 0.0 0.0
  B_y: 9.4356071469e-12 9.43866212262e-12 3.96315644851e-14 0.0042002134964
  B_z: 0.0 0.0 0.0

**Comment by rowanc1** [comment]
Is this identified by #367 ? Is "about" --> "exactly"? :)

**Comment by micmitch** [comment]
It is possible that these problems are related to the one identified in #367 but I think that they are separate problems. We see this problem with both E and B fields which use different averaging functions. Would have to look more carefully into the code for the field object. 

Not entirely sure what you mean by you question, is "about" --> "exactly"?

**Comment by rowanc1** [comment]
```
Comparing the E and B fields from field object vs. 
manual calculation from J and H show that the values 
differ by about 50%
```

If that is exactly 50% then I was thinking it might be the factor of 2 in the edge averaging code.

**Comment by rowanc1** [comment]
Is this all on the cyl mesh? If so, is the problem also on the 3D mesh?

**Comment by micmitch** [comment]
I see. I'll look into this a little more carefully. At the moment this is all from a cyl mesh. Will look at in on a 3D mesh to test X and Y oriented dipoles.

**Comment by lheagy** [comment]
@micmitch : can you confirm if this has been resolved?

**Comment by lheagy** [comment]
@micmitch: would you mind taking a look at this? I believe it should have been resolved

**Comment by lheagy** [comment]
This closed with #367. @micmitch, please re-open if you find the same issue.
